lib/logger/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+#[macro_use]
+extern crate log;
+
+use std::cell::RefCell;
+use std::env;
+use std::fs::File;
+use std::io::{self, Write};
+use std::sync::atomic;
+use std::thread;
+
+use log::{LevelFilter, Metadata, Record};
+
+static STATE: atomic::AtomicIsize = atomic::ATOMIC_ISIZE_INIT;
+
+thread_local! {
+    static WRITER: RefCell<Option<io::BufWriter<File>>> = RefCell::new(None);
+}
+
+pub fn initialize(filename_prefix: &str) {
+    // Ensure the thread local state is always properly initialized.
+    WRITER.with(|rc| {
+        if rc.borrow().is_none() {
+            rc.replace(Some(open_file(filename_prefix)));
+        }
+    });
+
+    let mut enable = false;
+    let level_filter = env::var_os("RUST_LOG").map(|val| {
+        enable = true;
+        match val.to_str().unwrap_or("") {
+            "off" => LevelFilter::Off,
+            "error" => LevelFilter::Error,
+            "warn" => LevelFilter::Warn,
+            "info" => LevelFilter::Info,
+            "debug" => LevelFilter::Debug,
+            "trace" => LevelFilter::Trace,
+            _ => {
+                eprintln!("unknown RUST_LOG value, defaulting to Trace");
+                LevelFilter::Trace
+            }
+        }
+    });
```

Comment:
This led me to peruse various docs, and I happened to notice the comment in here which gave me an idea:

https://docs.rs/env_logger/0.5.11/env_logger/struct.Builder.html#method.build

Would it work and make sense to nest env_logger or pretty_env_logger inside cranelift_logger? Then they could parse RUST_LOG for us, provide their filtering features, and so on.

If that doesn't work out well, that's ok, I just wanted to put the idea out there.

Reply:
I wondered about this too, I'll try tomorrow.